dev_tools/interactive/refinement_app/gsasIImodel.py

```diff
@@ -1,96 +1,96 @@
 import sys
+# import os
+import numpy as np
 sys.path.append("/srv/shiny-server/GSASII")
+sys.path.append("/home/mkscd/miniconda3/envs/GSASII/GSAS-II/GSASII")
 import GSASIIscriptable as G2sc  # type: ignore
-inputgpxfile = "infile.gpx"
 
-gpx = G2sc.G2Project(gpxfile=inputgpxfile, newgpx="output.gpx")
-gpx.save()
-# get the sample and instrument parameters from previous refinement in
-# dictionaries.
 
-# h.getHistEntryValue(['Sample Parameters', 'Type'], 'Bragg-Brentano')
-histnum = len(gpx.histograms())
-h = gpx.histograms()[0]
-
-sampleparams = {
-    "Scale": h.getHistEntryValue(['Sample Parameters', 'Scale']),
-    "DisplaceX": h.getHistEntryValue(['Sample Parameters', 'DisplaceX']),
-    "DisplaceY": h.getHistEntryValue(['Sample Parameters', 'DisplaceY']),
-    "Absorption": h.getHistEntryValue(['Sample Parameters', 'Absorption']),
-}
-sampreflist = []
-instreflist = []
-
-for param in sampleparams:
-    if sampleparams[param][1] is True:
-        sampreflist.append(param)
-print(h.getHistEntryValue(['Instrument Parameters'])[0]['Lam'])
-print(h.getHistEntryValue(['Instrument Parameters']))
-
-instdict = h.getHistEntryValue(['Instrument Parameters'])[0]
-# print tests
-# getinstdict = h.getHistEntryValue(['Instrument Parameters'])
-# print(getinstdict, "\n")
-# print(getinstdict["Zero"], "\n")
-
-
-instparams = {
-    "Lam": instdict['Lam'],
-    "Zero": instdict['Zero'],
-    "U": instdict['U'],
-    "V": instdict['V'],
-    "W": instdict['W'],
-    "X": instdict['X'],
-    "Y": instdict['Y'],
-    "Z": instdict['Z'],
-    # "SH/L": instdict['SH/L'],
-}
-print(instparams)
-
-for param in instparams:
-    if instparams[param][1] is True:
-        instreflist.append(param)
+def load_phase_constraints(gpx):
+    """
+    load the phase constraints previously added to the project
+    seems to not be readable
+    """
```

Comment:
## Formatting/style
Great to have docstrings - would be even better (IMHO) if they had args and returns sections. I think when combined with typehints it can really help make the docstring "write itself" - if you say what the arguments and return is, there's not much else to say about a function. Personally I normally use ["Google" style](https://github.com/NilsJPWerner/autoDocstring/blob/c9da64126fd9e667decd9d85b4e5b53c60372ea7/docs/google.md), but any formats fine.